### PR TITLE
changes for netmon4 - lastheard

### PR DESCRIPTION
**Pull request by DL2MF**
all files and changes for netmon4 - new lastheard page
(missing Makefile due to compile error and keyb.c with standard key assigns updated).

### CODE REVIEW

**Comment by sijskes** [comment]
build fails:
https://travis-ci.org/travisgoodspeed/md380tools/jobs/177496025

File: applet/src/lastheard.c [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/*
```
Comment (sijskes) [inline applet/src/lastheard.c]: why didnt you just use the syslog?

File: applet/src/keyb.c [hunk 2]
```diff
@@ -119,20 +128,17 @@ void handle_hotkey( int keycode )
             syslog_dump_dmesg();
             break ;
         case 7 :
```
Comment (sijskes) [inline applet/src/keyb.c]: dont reorder keys, just because you have added 1 option.

File: applet/src/lastheard.c (hunk 1 above)
Comment (DL2MF) [inline applet/src/lastheard.c]: currently lh already uses it's own buffer and logarea. my first try was using syslog, but didn't get it work asap, so I created a clone named lastheard. bringing both together may be a later goal for me, it needs some more review on syslog, to enhance this to handle different logbuffers.

File: applet/src/keyb.c (hunk 2 above)
Comment (DL2MF) [inline applet/src/keyb.c]: I asked you to give me some feedback about the keys in https://groups.google.com/forum/#!topic/md380tools/NYdVOy0NI0c
You asked me just to do a PR ... may be I misunderstood this, feel free to let me feel like a fool.

Comment (sijskes) [inline applet/src/keyb.c]: look. if you dont present the code, it cannot be evaluated. dont whine about it.

**Comment by DL2MF** [comment]
No, be sure I'm glad to share my small part of work to this great project, without it I won't have any idea to bring to this. My only question is which (new) key should be used for netmon4 and do you accept, that we have this two additional files lastheard.c / lastheard.h? If this is a showstopper, I need the break to improve my ideas ...
What do you prefer about the keyb.c? Start a complete new PR or make a seperate for keyb.c with the old netmon keys on the 7,8,9,# in the same order before and a new key for the lastheard ...?

**Comment by sijskes** [comment]
Just modify the branch you did the PR on. The build will automatically starts rebuilding.
You have to fix the compile errors also. Fix the makefile and commit and push to your own repo. That should work.

File: applet/src/radiostate.c [hunk 3]
```diff
@@ -59,7 +59,8 @@ void rst_voice_lc_header(lc_t *lc)
         
         LOGR("cs %c %d->%d\n", grp_c, src, dst );
 
-        rst_voice_active = 1 ;    
+        rst_voice_active = 1 ;
+	rx_voice = 1 ;				// flag for new voice call received
```
Comment (sijskes) [inline applet/src/radiostate.c]: why did you introduce flag with the exact same purpose?

File: applet/src/lastheard.c (hunk 1 above)
Comment (sijskes) [inline applet/src/lastheard.c]: what did you get working asap?

File: applet/src/keyb.c [hunk 4]
```diff
@@ -148,7 +157,7 @@ void handle_hotkey( int keycode )
             //mb_send_beep(beep_event_probe);
             break ;
         case 15 :
-            syslog_redraw();
+            bp_send_beep(BEEP_TEST_2);
```
Comment (sijskes) [inline applet/src/keyb.c]: why did you remove the syslog_redraw

File: applet/src/radiostate.c (hunk 3 above)
Comment (DL2MF) [inline applet/src/radiostate.c]: rst_voice_active is being set during the whole reception of a voice call, using this would cause multiple lheard entries in the list for the whole call hold time during reception. rx_voice is set the same time, but is being reset immediately after the first lh logentry has been written to the display/buffer.

File: applet/src/keyb.c (hunk 4 above)
Comment (DL2MF) [inline applet/src/keyb.c]: maybe a copy&paste error when moving key # back to it's original function, can't check now, have to look into my sources tonight.
